hedera-node/src/main/java/com/hedera/services/fees/calculation/contract/queries/ContractCallLocalResourceUsage.java

```diff
@@ -75,7 +77,8 @@ public ContractCallLocalResourceUsage(
 			final CallLocalEvmTxProcessor evmTxProcessor,
 			final EntityIdSource ids,
 			final OptionValidator validator,
-			final AliasManager aliasManager
+			final AliasManager aliasManager,
+			final SideEffectsTracker sideEffectsTracker
```

Comment:
This object will be used by query-answering threads from the Netty thread pool.

It's **critical** to prevent any query thread from using a stateful `@Singleton` object also used by `handleTransaction()`; and the `sideEffectsTracker` here is just such an object. 

So we should construct a new `SideEffectsTracker` for each `HederaWorldState` instance created on line 123. (Though notice that since `ContractCallLocal` performs a "static" EVM call, it **cannot** have any side effects! So it might also be reasonable to have a `HederaWorldState` constructor that doesn't take a `sideEffectsTracker`; but probably not worth the effort right now.)